chain/chain/src/resharding.rs

```diff
@@ -205,6 +206,15 @@ impl Chain {
         Ok(())
     }
 
+    /// Function to check whether the snapshot is ready for resharding or not. We return true if the snapshot is not
+    /// ready and we need to retry/reschedule the resharding job.
+    pub fn retry_build_state_for_split_shards(state_split_request: &StateSplitRequest) -> bool {
```

Comment:
Nit: Maybe return an enum with some meaningful values like SnapshotReady, SnapshotMissing?

Reply:
I feel the boolean is a better option here (although in usual cases enums are the way to go) and we are modeling this as a direct condition in our state sync actor, i.e. whether we should retry resharding or not. In the foreseeable future, I don't see us adding a different enum here.

Let me know if you'd still like me to change this to an enum.